### PR TITLE
COMMON-003 exchange readiness contract

### PR DESCRIPTION
**Pull request by pivox**
## Summary
- add the pure `App\Exchange\Readiness` contract: readiness levels, input/report DTOs, evaluator, and runtime check interface
- enforce fail-closed readiness rules for missing instruments, missing mainnet guard, missing SL capability, kill switch, and demo/testnet enablement
- document the COMMON-003 readiness levels and report shape in the exchange readiness matrix

## Scope
- no REST, WebSocket, Doctrine, Messenger, Temporal, or runtime trading calls
- no mainnet/live readiness level
- no strategy, EntryZone, Risk/Leverage, SL/TP behavior changes

## References
- COMMON-003 — Exchange readiness contract
- Depends on COMMON-002 / PR #222

## Tests
- `vendor/bin/phpunit tests/Exchange/Readiness tests/TradingCore/Config/EffectiveTradingConfigRuntimeFilesTest.php`
- `vendor/bin/phpstan analyse src/Exchange/Readiness tests/Exchange/Readiness --memory-limit=1G`
- `php bin/console lint:container --no-debug` (passes; existing PHP 8.4 deprecation from `EntryZoneStatsCommand`)
- `python3 -m mkdocs build --strict`
- `git diff --cached --check`

### CODE REVIEW

**Comment by pivox** [comment]
@codex please review this PR. Focus on correctness, regressions, readiness level semantics, redaction, architecture boundaries, missing tests, and edge cases. Please post review threads on specific lines when useful, or a PR comment if the feedback is general.

File: trading-app/src/Exchange/Readiness/ExchangeReadinessReport.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+declare(strict_types=1);
+
+namespace App\Exchange\Readiness;
+
+use App\Common\Enum\Exchange;
+use App\Common\Enum\MarketType;
+
+final readonly class ExchangeReadinessReport
+{
+    private const SENSITIVE_PATTERN = '/(api[_-]?key|secret|private[_-]?key|passphrase|authorization|cookie|token|signature|credentials?)/i';
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Exchange/Readiness/ExchangeReadinessReport.php]: Fixed in bb7708b. The report redaction pattern now covers `password`, `memo`, and `sign` in addition to the previous secret labels, so messages such as `password=...`, `BITMART_API_MEMO=...`, and `OK-ACCESS-SIGN=...` are redacted in `toArray()`. Added regression coverage for those labels.

File: trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    private function hasPrivateRead(ExchangeReadinessInput $input): bool
+    {
+        return $input->privateReadConnectivity
+            && $input->accountReadable
+            && $input->permissionsRead;
+    }
+
+    private function hasLocalDryRunReadiness(ExchangeReadinessInput $input): bool
+    {
+        return $input->dryRun
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php]: Fixed in bb7708b. The evaluator no longer requires `dryRun=true` for the guarded readiness prerequisites. `dryRun=false` now reaches `demo_testnet_enabled` when the environment is demo/testnet, demo/testnet writes are explicitly enabled, kill switch is off, trade permission is present, and the other guards pass. Added regression coverage for this path.

File: trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            return $this->report($input, $readyLevel, [], $warnings);
+        }
+
+        if (!$input->permissionsTrade) {
+            $warnings[] = 'trade_permission_required_for_demo_testnet_enabled';
+
+            return $this->report($input, $readyLevel, [], $warnings);
+        }
+
+        return $this->report($input, ExchangeReadinessLevel::DemoTestnetEnabled, [], $warnings);
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php:97]: Fixed in bb7708b. `demo_testnet_enabled` is now gated to `environment=demo|testnet`; a `dryRun=false` request outside those environments returns `not_ready` with `demo_testnet_environment_required`. Candidate readiness is also kept below demo/testnet levels outside demo/testnet environments. Added regression coverage.

**Comment by pivox** [comment]
@codex review

File: trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        return $input->privateReadConnectivity
+            && $input->accountReadable
+            && $input->permissionsRead;
+    }
+
+    private function hasGuardedReadiness(ExchangeReadinessInput $input): bool
+    {
+        return $input->demoTestnetWriteGuard
+            && ($input->allowedSymbols !== [] || $input->allowedMarkets !== [])
+            && $input->maxNotional !== null;
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php:141]: Fixed in 556efda. The evaluator now normalizes `allowedSymbols` and `allowedMarkets` before using them as guarded-readiness prerequisites and before putting them on the report. Non-string values and blank strings after trim are dropped, duplicates are collapsed, and an empty normalized whitelist keeps the report at `private_read_only` with `local_dry_run_prerequisites_missing`. Added regression coverage for blank whitelist entries.

**Comment by pivox** [comment]
@codex review

File: trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    private function hasGuardedReadiness(ExchangeReadinessInput $input): bool
+    {
+        return $input->demoTestnetWriteGuard
+            && ($this->normalizedList($input->allowedSymbols) !== [] || $this->normalizedList($input->allowedMarkets) !== [])
+            && $input->maxNotional !== null;
+    }
+
+    private function isDemoOrTestnet(string $environment): bool
+    {
+        return in_array(strtolower($environment), ['demo', 'testnet'], true);
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php:155]: Fixed in be57f77. The evaluator now requires the same supported pairs as COMMON-002 before promoting to demo/testnet levels: OKX/demo or Hyperliquid/testnet. A `dryRun=false` cross-pair returns `not_ready` with `exchange_environment_pair_unsupported`, and dry-run cross-pairs remain below demo/testnet candidate. Added regression coverage for Hyperliquid/demo.

**Comment by pivox** [comment]
@codex review

File: trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    {
+        return $input->privateReadConnectivity
+            && $input->accountReadable
+            && $input->permissionsRead;
+    }
+
+    private function hasGuardedReadiness(ExchangeReadinessInput $input): bool
+    {
+        return $input->demoTestnetWriteGuard
+            && ($this->normalizedList($input->allowedSymbols) !== [] || $this->normalizedList($input->allowedMarkets) !== [])
```
[the remark this answers is not in the document]
Comment (pivox) [reply trading-app/src/Exchange/Readiness/ExchangeReadinessEvaluator.php]: Fixed in d72ade2. `allowedMarkets` now satisfies guarded readiness only when it contains the checked `$input->marketType->value`; otherwise the evaluator requires a usable symbol whitelist. Added regression coverage for `marketType=perpetual` with only `allowedMarkets=[spot]`, which now stays at `private_read_only` with `local_dry_run_prerequisites_missing`.

**Comment by pivox** [comment]
@codex review
